msm/src/circuit_design/witness.rs

```diff
@@ -43,6 +43,9 @@ pub struct WitnessBuilderEnv<
     /// value for row #j of the selector #i.
     pub fixed_selectors: Vec<Vec<F>>,
 
+    /// Function used to map assertions.
+    pub assert_mapper: Box<dyn Fn(F) -> F>,
```

Comment:
I am struggling understanding the real advantage of this, and I am not sure I fully understand it for now.

Reply:
Fixed, got it.